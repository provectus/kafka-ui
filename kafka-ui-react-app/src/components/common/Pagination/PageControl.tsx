import React from 'react';

import { PaginationLink } from './Pagination.styled';

export interface PageControlProps {
  current: boolean;
  url: string;
  page: number;
}

const PageControl: React.FC<PageControlProps> = ({ current, url, page }) => {
  return (
    <li>
      <PaginationLink
        to={url}
        aria-label={`Goto page ${page}`}
        $isCurrent={current}
        role="button"
      >
        {page}
      </PaginationLink>
    </li>
  );
};

export default PageControl;
