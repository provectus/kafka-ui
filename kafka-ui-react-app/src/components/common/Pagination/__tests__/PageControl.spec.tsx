import React from 'react';
import { mount, shallow } from 'enzyme';
import { StaticRouter } from 'react-router';
import PageControl, { PageControlProps } from '../PageControl';

const page = 138;

describe('PageControl', () => {
  const setupWrapper = (props: Partial<PageControlProps> = {}) => (
    <StaticRouter>
      <PageControl url="/test" page={page} current {...props} />
    </StaticRouter>
  );

  it('renders current page', () => {
    const wrapper = mount(setupWrapper({ current: true }));
    expect(wrapper.exists('.pagination-link.is-current')).toBeTruthy();
  });

  it('renders non-current page', () => {
    const wrapper = mount(setupWrapper({ current: false }));
    expect(wrapper.exists('.pagination-link.is-current')).toBeFalsy();
  });

  it('renders page number', () => {
    const wrapper = mount(setupWrapper({ current: false }));
    expect(wrapper.text()).toEqual(String(page));
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<PageControl url="/test" page={page} current />);
    expect(wrapper).toMatchSnapshot();
  });
});
