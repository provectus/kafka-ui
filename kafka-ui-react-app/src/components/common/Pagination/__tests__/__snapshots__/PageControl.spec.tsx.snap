// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageControl matches snapshot 1`] = `
<li>
  <Link
    aria-label="Goto page 138"
    className="pagination-link is-current"
    to="/test"
  >
    138
  </Link>
</li>
`;
